### PR TITLE
Updated documentation

### PR DESCRIPTION
**Pull request by Vaydheesh**
- updated `Readme.md` and `docs/` 
- use wrappers from apertium repository #44 #47 
- Makefile for pip releases #66

### CODE REVIEW

**Comment by ftyers** [comment]
Just an aside: Should it be `apertium-python` or `python-apertium`? Most packages I see in Debian are `python-...`.

**Comment by Vaydheesh** [comment]
I don't know why, but the `apt-get install` is returning exit code 100, even after installing `apt-transport-https`. https://travis-ci.com/apertium/apertium-python/builds/125114987

- The build status isn't consistent for same commit

https://travis-ci.com/apertium/apertium-python/builds/125094062
https://travis-ci.com/apertium/apertium-python/builds/125109465

**Comment by sushain97** [comment]
I think it was an intermittent issue? You might be able to revert the other commit now.

**Comment by Vaydheesh** [comment]
Should I the retry the installation process before exiting?

**Comment by sushain97** [comment]
Hmmm, not sure what you mean?

**Comment by Vaydheesh** [comment]
Retry https://github.com/apertium/apertium-python/blob/master/apertium/installer.py#L113-L115, if return code isn't 0

**Comment by sushain97** [comment]
Nah, I don't think that should be our default.

File: apertium/translation/__init__.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def translate(self, text: str, mark_unknown: bool = False, formatting: Optional[
             return result.decode()  # type: ignore
 
 
-def translate(l1: str, l2: str, text: str, mark_unknown: bool = False, formatting: Optional[str] = None, deformat: str = 'txt', reformat: str = 'txt') -> str:
+def translate(l1: str, l2: str, text: str, mark_unknown: bool = False, formatting: Optional[str] = None, deformat: str = 'txt', reformat: str = 'txt') -> Any:
```
Comment (sushain97) [inline apertium/translation/__init__.py]: Why is this `Any`?

File: docs/source/about.md [hunk 2]
```diff
@@ -4,18 +4,18 @@ About
...
+- The existing codebase has `Subprocess` and SWIG wrapper implementations of the basic functions used by Apertium modules.
 
-Contribute
-----------
+Contributing
+------------
 
 - Issue Tracker: <https://www.github.com/apertium/apertium-python/issues>
 - Source Code: <https://www.github.com/apertium/apertium-python>
+- Run unit tests with `python setup.py test`.
```
Comment (sushain97) [inline docs/source/about.md]: We should `chmod +x setup.py` so this can just be `setup.py test`.

File: docs/source/install.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+Installation
+============
+
+From PyPI
+-----------
+
+- Install from PyPI by running
+
+
```
Comment (sushain97) [inline docs/source/install.md]: Extra newline here.

File: docs/source/install.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    pip install apertium-python
+
+- For developers
+
+
+    git clone https://github.com/apertium/apertium-python.git
```
Comment (sushain97) [inline docs/source/install.md]: If you look at the rich diff on Github's change view, these are not rendering as code blocks:

image

second button here.

File: setup.py [hunk 5]
```diff
@@ -20,16 +24,31 @@ def _post_install():
...
-    name='apertium-python',
-    # TODO: Add version description, keywords, author, author_email
-    license='GNU General Public License v3.0 ',
+    name='apertium',
+    author=find_details("author", ["apertium", "__init__.py"]),
+    author_email='sushain@skc.name',
+    license=find_details("license", ["apertium", "__init__.py"]),
+    version=find_details("version", ["apertium", "__init__.py"]),
+    keywords='Apertium Machine Translation Linguistics',
+    description='This project is an attempt to make the Apertium Core modules available in python',
```
Comment (sushain97) [inline setup.py]: We can do better than this :)

File: setup.py [hunk 6]
```diff
@@ -20,16 +24,31 @@ def _post_install():
...
 setup(
-    name='apertium-python',
-    # TODO: Add version description, keywords, author, author_email
-    license='GNU General Public License v3.0 ',
+    name='apertium',
+    author=find_details("author", ["apertium", "__init__.py"]),
+    author_email='sushain@skc.name',
+    license=find_details("license", ["apertium", "__init__.py"]),
+    version=find_details("version", ["apertium", "__init__.py"]),
+    keywords='Apertium Machine Translation Linguistics',
```
Comment (sushain97) [inline setup.py]: These should all be lowercase.

File: setup.py [hunk 7]
```diff
@@ -20,16 +24,31 @@ def _post_install():
...
+    description='This project is an attempt to make the Apertium Core modules available in python',
     long_description=open(path.join(path.abspath(path.dirname(__file__)), 'README.md')).read(),
     long_description_content_type='text/markdown; charset=UTF-8',
+    platforms=['Ubuntu', 'Windows'],
     url='https://github.com/apertium/apertium-python',
```
Comment (sushain97) [inline setup.py]: We need to add a bunch of classifiers: https://github.com/apertium/streamparser/blob/master/setup.py#L17

**Comment by sushain97** [comment]
Going to merge this so I can release to PyPi and let you bring in a different PR that addresses the remaining concerns.

**Comment by sushain97** [comment]
Released @ https://pypi.org/project/apertium/  !

**Comment by sushain97** [comment]
@Vaydheesh can you send a release announcement email to the mailing list saying that it's available for trying out in an alpha stage via `pip install apertium` and to file any suggestions/issues in the issue tracker on GitHub?

**Pull request merged**